store/populations.go

```diff
@@ -39,18 +40,44 @@ type PopObs struct {
 func (s *store) GetPopObs(ctx context.Context, in *pb.GetPopObsRequest,
 	out *pb.GetPopObsResponse) error {
 	dcid := in.GetDcid()
+	key := util.BtPopObsPrefix + dcid
 
-	btRow, err := s.btTable.ReadRow(ctx, util.BtPopObsPrefix+dcid)
+	res1 := pb.PopObsPlace{}
+	if content, ok := s.cacheData[key]; ok {
+		c, err := util.UnzipAndDecode(content)
+		if err != nil {
+			log.Print(err)
+		} else {
+			jsonpb.UnmarshalString(string(c), &res1)
+		}
+	}
+
+	if res1.Populations == nil {
+		res1.Populations = map[string]*pb.PopObsPop{}
+	}
+
+	res2 := pb.PopObsPlace{}
+	btRow, err := s.btTable.ReadRow(ctx, key)
 	if err != nil {
 		return err
 	}
-	// TODO(boxu): merge the gcs data with main cache.
 	if len(btRow[util.BtFamily]) > 0 {
-		out.Payload = string(btRow[util.BtFamily][0].Value)
-	} else {
-		out.Payload, _ = util.ZipAndEncode("{}")
+		c, err := util.UnzipAndDecode(string(btRow[util.BtFamily][0].Value))
+		if err != nil {
+			log.Print(err)
+		}
+		jsonpb.UnmarshalString(string(c), &res2)
+
+		for k, v := range res2.Populations {
+			res1.Populations[k] = v
+		}
```

Comment:
To clarify, if there is an overlapping pop-id, here, we are picking the BT one?

I wonder if we should pick the side-cache?

Reply:
Changed so side cache has higher priority